### PR TITLE
I hope i dint brake anything

### PR DESCRIPTION
**Pull request by BastelPichi**

### CODE REVIEW

**Comment by BastelPichi** [comment]
Im kinda retarded dont merge this yet lmfao

**Comment by BastelPichi** [comment]
Dunno if this works xD
Should tho

File: src/index.ts [hunk 1]
```diff
@@ -68,11 +67,19 @@ app.get(`/:id`, (request, response) => {
 app.post(`/`, (request, response) => {
     // Check if it is a valid url
     const { url } = request.body;
-    if (!url || !config.urlRegex.test(url)) {
+    
+    try {
+        url = new URL(string);
```
Comment (sys-256) [inline src/index.ts:72]: You're overwriting the `url` variable here. Just `new URL(string);` should be fine too, as that should throw an error when an invalid URL is provided too (not sure, just assuming this).

File: src/index.ts [hunk 2]
```diff
@@ -68,11 +67,19 @@ app.get(`/:id`, (request, response) => {
...
+    } catch (_) {
```
Comment (sys-256) [inline src/index.ts:73]: Can be replaced with `} catch {` or `} catch () {`, not sure which one it was again. Just see for which one VSCode throws an error.

File: src/index.ts [hunk 3]
```diff
@@ -68,11 +67,19 @@ app.get(`/:id`, (request, response) => {
 app.post(`/`, (request, response) => {
     // Check if it is a valid url
     const { url } = request.body;
-    if (!url || !config.urlRegex.test(url)) {
+    
+    try {
```
Comment (sys-256) [inline src/index.ts:71]: I think something with `new URL(string).catch(() => {respond with HTTP 400});` should work here, but i'll try myself soonish.

File: src/index.ts [hunk 4]
```diff
@@ -68,11 +67,19 @@ app.get(`/:id`, (request, response) => {
...
+        url = new URL(string);
+    } catch (_) {
         response.status(400).send(`Please specify a (valid) URL.`);
         return;
     }
 
+    if (!url.protocol === "http:" || !url.protocol === "https:")  {
```
Comment (sys-256) [inline src/index.ts:78]: I think protocols like FTP should work too, so we can remove this `if` statement.

**Review by sys-256 — CHANGES_REQUESTED** [review]
I'm planning on updating this myself, but if i haven't yet in a few days, you are free to update it yourself too, and then i'll merge the changes.

File: src/index.ts (hunk 4 above)
Comment (BastelPichi) [reply src/index.ts:78]: No. `javascript:void(0)` World be valid the, too. However you can still just add ftp to the if.

File: src/index.ts (hunk 3 above)
Comment (BastelPichi) [reply src/index.ts:71]: Do i look live i know ts? Man, thats directly from stackoverflow lmfao...

File: src/index.ts (hunk 1 above)
Comment (BastelPichi) [reply src/index.ts:72]: Yes, that should work.

**Comment by BastelPichi** [comment]
If you fix IT yourself please gib me the Code so im the contrib lmfao

**Comment by BastelPichi** [comment]
Ahlo?

**Comment by sys-256** [comment]
Running `npm i` gives some errors on the laptop I'm using rn (PC broke) so I can't develop it rn.

**Comment by BastelPichi** [comment]
k, imma actually test this, then maybe create another pr. (maybe)

**Pull request closed without merging**